File: class/toast&dialog/app/src/main/java/com/example/toastdialog/MainActivity.java
package com.example.toastdialog;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import android.app.appsearch.SearchResult;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btn_showdialog = findViewById(R.id.btn_ShowDialog);

        btn_showdialog.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);

                builder.setTitle("請選擇功能");
                builder.setMessage("請根據下方按鈕選擇要顯示的物件");

                builder.setPositiveButton("顯示List", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        showListDialog();
                    }
                });

                builder.setNegativeButton("自定義Toast", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        showToast();
                    }
                });

                builder.setNeutralButton("取消", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        Toast.makeText(MainActivity.this, "關閉Dialog", Toast.LENGTH_SHORT).show();
                    }
                });

                builder.show();

            }
        });

    }

    private void showToast(){
        Toast toast = new Toast(MainActivity.this);
        toast.setGravity(Gravity.TOP, 0, 50);
        toast.setDuration(Toast.LENGTH_SHORT);

        LayoutInflater inflater = getLayoutInflater();
        View view = inflater.inflate(R.layout.custom_toast, findViewById(R.id.custom_toast_root));

        toast.setView(view);
        toast.show();
    }

    private void showListDialog(){
        final String list[] = {"Message1", "Message2", "Message3", "Message4`", "Message5"};
        AlertDialog.Builder builder = new AlertDialog.Builder(MainActivity.this);
        builder.setTitle("使用List呈現");
        builder.setItems(list, new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                Toast.makeText(MainActivity.this, "你選擇的是" + list[i], Toast.LENGTH_SHORT).show();
            }
        });

        builder.show();
    }

}

File: homework/KT02/ch06-toast&dialog/toast/app/src/main/java/com/example/toast/MainActivity.kt
package com.example.toast

import android.content.DialogInterface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AlertDialog

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var btn_showdialog = findViewById<Button>(R.id.btn_ShowDialog)

        btn_showdialog.setOnClickListener {

            var mDialog = AlertDialog.Builder(this)

            mDialog.setTitle("請選擇功能")
            mDialog.setMessage("請根據下方按鈕選擇要顯示的物件")

            mDialog.setNeutralButton("取消") { _, _ ->
                Toast.makeText(
                    this,
                    "Dialog關閉",
                    Toast.LENGTH_SHORT
                ).show()
            }

            mDialog.setNegativeButton("自定義Toast"){ _, _ ->
                showToast()
            }

            mDialog.setPositiveButton("顯示List"){ _, _ ->
                showListDialog()
            }

            mDialog.show()
        }

    }


    private fun showToast() {
        var toast = Toast(this)
        toast.setGravity(Gravity.TOP, 0, 150)
        toast.duration = Toast.LENGTH_SHORT

        var inflater = getLayoutInflater()
        var layout = inflater.inflate(R.layout.custom_toast, findViewById(R.id.custom_toast_root))
        toast.view = layout

        toast.show()
    }

    private fun showListDialog(){
        val list = arrayOf("Message1", "Message2", "Message3", "Message4", "Message5")
        val dialog_list = AlertDialog.Builder(this)

        dialog_list.setTitle("使用List呈現")

        dialog_list.setItems(list) {_, i ->
            Toast.makeText(this,  "你選的是" + list[i], Toast.LENGTH_SHORT).show()
        }

        dialog_list.show()
    }
}